realsense2_camera/src/base_realsense_node.cpp

```diff
@@ -2474,3 +2486,480 @@ void TemperatureDiagnostics::diagnostics(diagnostic_updater::DiagnosticStatusWra
         status.summary(0, "OK");
         status.add("Index", _crnt_temp);
 }
+
+/****************************** NOMAGIC ******************************/
+
+struct Clock
+{
+    using InternalClock = std::chrono::steady_clock;
+
+    Clock() : start(InternalClock::now()) {}
+    void restart() {
+        start = InternalClock::now();
+    }
+
+    double getElapsedSecs() {
+        return static_cast<std::chrono::duration<double>>(InternalClock::now() - start).count();
+    }
+
+private:
+    InternalClock::time_point start;
+};
+
+void BaseRealSenseNode::nomagicSetupService(stream_index_pair stream, bool is_aligned_depth)
+{
+    // name + [index]: color, depth, infra1, infra2, etc.
```

Comment:
Please add a bit more context to this comment. Is this about what the STREAM_NAME macro will produce?